packages/vc-verification/src/issuer-def-resolver.ts

```diff
@@ -0,0 +1,54 @@
+import { IIssuerDefinition } from '@energyweb/credential-governance';
+import { providers, utils } from 'ethers';
+import { RoleDefinitionResolverV2__factory } from '@energyweb/credential-governance/ethers/factories/RoleDefinitionResolverV2__factory';
+import { RoleDefinitionResolverV2 } from '@energyweb/credential-governance/ethers/RoleDefinitionResolverV2';
+
+/**
+ * An interface for Resolution of issuers for a namespace
+ */
+export interface IssuerDefinitionResolver {
+  /**
+   * Fetches authorised issuers for the provided namespace
+   * @param namespace
+   * @returns IIssuerDefinition for the namespace
+   */
+  getIssuerDefinition(namespace: string): Promise<IIssuerDefinition>;
+}
+
+/**
+ * Resolves an issuer definition by reading smart contract state via an Ethers provider
+ */
+export class EthersProviderIssuerDefinitionResolver
+  implements IssuerDefinitionResolver
+{
+  private _roleDefResolver: RoleDefinitionResolverV2;
+
+  constructor(provider: providers.Provider, roleDefResolverAddr: string) {
+    this._roleDefResolver = RoleDefinitionResolverV2__factory.connect(
+      roleDefResolverAddr,
+      provider
+    );
+  }
+
+  /**
+   *
+   * @param namespace
+   * @returns issuers for the namespace from blockchain contract
+   */
+  async getIssuerDefinition(namespace: string): Promise<IIssuerDefinition> {
+    const issuers: IIssuerDefinition = {};
+    const result = await this._roleDefResolver.issuers(
+      namespace.startsWith('0x') ? namespace : utils.namehash(namespace)
+    );
+    const type = await this._roleDefResolver.issuerType(
+      namespace.startsWith('0x') ? namespace : utils.namehash(namespace)
+    );
```

Comment:
Could we wrap it with `Promise.all` to increase performance?

Reply:
Do you think `fail-first` reason for using `Promise.all`? IMO the above case should always be consistent in any scenarios, so the performance remains same?

Reply:
`Promise.all()` will run it at the same moment and not in order. In this case we are blocking thread just once and not twice. 
Second problem here is: we are passing the same argument maybe we can do smth like `const resolvedNamespace = namespace.startsWith('0x') ? namespace : utils.namehash(namespace)` and then pass to those functions `resolvedNamespace`